packages/eas-cli/src/commands/branch/publish.ts

```diff
@@ -198,10 +205,25 @@ export default class BranchPublish extends Command {
         updatesToRepublish = updates.filter(update => update.group === updateGroup);
       }
 
+      if (updatesToRepublish.length === 0) {
+        throw new Error(`There are no updates in this group`);
+      }
+
       for (const update of updatesToRepublish) {
         const { platform, manifestFragment } = update;
-        updateInfoGroup[platform as PublishPlatforms] = JSON.parse(manifestFragment);
+
+        // If the platformFlag is not set, add all platforms. If it is set, only add the specified platform
+        if (!platformFlag) {
+          updateInfoGroup[platform as PublishPlatform] = JSON.parse(manifestFragment);
```

Comment:
assert instead of cast would be safer, but I don't now that API so I'm not sure if it's worth doing